data/shared/citizen/scripting/v8/timer.js

```diff
@@ -91,10 +93,22 @@
             i = tickers.length;
 
             while (i--) {
+                const ticker = tickers[i];
+
                 try {
-                    tickers[i]();
-                } catch(e) {
-                    console.error('Unhandled error: ' + e.toString() + '\n' + e.stack);
+                    if (ticker instanceof generatorConstructor) {
```

Comment:
I'd rather suggest to have different array for generator tickers, i.e. `setTick` would distinguish generators from other functions and put them in different array. Of course that would require another `while` loop down there, but advantage is that we can bail out of generators execution early. We're already doing it like so for animation frames.